config/src/test/kotlin/org/springframework/security/config/web/servlet/SessionManagementDslTests.kt

```diff
@@ -29,31 +34,76 @@ import org.springframework.security.config.annotation.web.configuration.WebSecur
 import org.springframework.security.web.authentication.session.SessionAuthenticationStrategy
+import org.springframework.security.web.context.HttpSessionSecurityContextRepository
+import org.springframework.security.web.csrf.CsrfToken
+import org.springframework.security.web.csrf.HttpSessionCsrfTokenRepository
+import org.springframework.security.web.savedrequest.HttpSessionRequestCache
+import org.springframework.security.web.savedrequest.SavedRequest
 import org.springframework.security.web.session.SimpleRedirectInvalidSessionStrategy
 import org.springframework.test.web.servlet.MockMvc
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers.redirectedUrl
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
-import javax.servlet.http.HttpServletRequest
-import javax.servlet.http.HttpServletResponse
+import org.springframework.web.servlet.support.SessionFlashMapManager
 
 /**
  * Tests for [SessionManagementDsl]
  *
  * @author Eleftheria Stein
  */
 class SessionManagementDslTests {
+
+    companion object {
+        private const val SAVED_REQUEST_ATTR_NAME = "SPRING_SECURITY_SAVED_REQUEST"
+
+        private val DEFAULT_CSRF_TOKEN_ATTR_NAME = "${HttpSessionCsrfTokenRepository::class.java.name}.CSRF_TOKEN"
+        private val FLASH_MAPS_SESSION_ATTR_NAME = "${SessionFlashMapManager::class.java.name}.FLASH_MAPS"
+    }
+
     @Rule
     @JvmField
     val spring = SpringTestRule()
 
     @Autowired
     lateinit var mockMvc: MockMvc
 
+    private val mockedCsrfToken: CsrfToken = mockk(relaxed = true)
+    private val mockedSavedRequest: SavedRequest = mockk(relaxed = true)
+    private val mockedSecurityContext: SecurityContext = mockk(relaxed = true)
+    private val mockedAuthentication: Authentication = mockk(relaxed = true)
+    private val session: MockHttpSession = mockk()
+
+    @Before
+    fun setUp() {
+        every { session.changeSessionId() } throws SessionAuthenticationException("any SessionAuthenticationException")
+        every { session.access() } returns Unit
+        every { session.setAttribute(any(), any()) } returns Unit
+        every { session.isInvalid } returns false
+        every { session.id } returns "any id"
+        every { session.getAttribute(DEFAULT_CSRF_TOKEN_ATTR_NAME) } returns mockedCsrfToken
+        every { session.removeAttribute(DEFAULT_CSRF_TOKEN_ATTR_NAME) } returns Unit
+        every { session.getAttribute(SAVED_REQUEST_ATTR_NAME) } returns mockedSavedRequest
+        every { session.removeAttribute(SAVED_REQUEST_ATTR_NAME) } returns Unit
+        every { session.getAttribute(FLASH_MAPS_SESSION_ATTR_NAME) } returns emptyList<Any>()
+        every { session.removeAttribute(FLASH_MAPS_SESSION_ATTR_NAME) } returns Unit
+        every {
+            session.getAttribute(HttpSessionSecurityContextRepository.SPRING_SECURITY_CONTEXT_KEY)
```

Comment:
This line is what is causing the test `session management when session authentication error url then redirected to url` to fail. The `SPRING_SECURITY_CONTEXT_KEY` attribute must be null in order to reach the code block that causes the authentication error.
https://github.com/spring-projects/spring-security/blob/68f91edbb8a223ee8505893697d85cbe51890ea2/web/src/main/java/org/springframework/security/web/session/SessionManagementFilter.java#L91-L105
We may not be able to use Mockk in this case, because I believe it doesn't support returning null values.

Reply:
Yea, you are right. Mockk doesn't support `null` as a returning type. This makes really sense, because kotlin is designed for the null-saftey. I rollbacked the  testclass `SessionManagementDslTests`, so that Mockito is used.
Thank you very much for the help.